.github/workflows/ci.yml

```diff
@@ -69,8 +69,18 @@ jobs:
       - name: Setup OS matrix
         id: setup-os-matrix
         run: |
-          os='["ubuntu-latest","windows-latest","macos-latest"'
+          os='["ubuntu-latest","windows-latest"'
+
+          # runs on macOS only if there is a push to master, or a tag is 
+          # pushed, we do this since macOS builds last too long and ILGPU 
+          # is rarely used on a macOS
+          [ "${{ github.ref }}" == "refs/heads/master" ] ||
+          [[ "${{ github.ref }}" =~ "refs/tags/v" ]] && 
+          [ "${{ github.event_name }}" == "push" ] &&
+          os="$os,\"macos-latest\""
```

Comment:
should we also add here other platforms? if all conditions are met, we want to run for all platforms (previous state), not just macOS? 🤔

Reply:
ye, it basically appends to initial `os='["ubuntu-latest","windows-latest"'` so in case this part gets evaluated it would be `os='["ubuntu-latest","windows-latest","macos-latest"'`

Reply:
oh, my bad i forgot ` os="$os,\"macos-latest\"` at beginning 
i saw onyl `macos-latest` 🤣 
looks good 👌